lib/ebae_web/controllers/buy_controller.ex

```diff
@@ -1,7 +1,56 @@
 defmodule EbaeWeb.BuyController do
   use EbaeWeb, :controller
 
+  alias Ebae.{Auctions, Auctions.Bid}
+  alias EbaeWeb.Auth
+
   def index(conn, _) do
     render(conn, "index.html")
   end
+
+  def new(conn, %{"id" => auction_id}) do
+    render(conn, "new.html",
+      changeset: Auctions.change_bid(%Bid{}),
+      action: Routes.buy_path(conn, :create, auction_id)
+    )
+  end
+
+  def create(conn, %{"bid" => bid, "id" => auction_id}) do
+    conn
+    |> validate_and_create(bid, auction_id)
+    |> create_reply(conn, auction_id)
+  end
+
+  defp validate_and_create(conn, bid, auction_id) do
+    with true <- validate(bid),
```

Comment:
Is there a benefit to using the `with` here outside of checking the result of `validate`? I feel like this might be more clear if you used a `case` statement (or an `if` if you prefer)

```elixir
if validate(bid)
   # something
 else
   :error
end
`````

Reply:
I feel like it is a smell there is  not (much) pattern matching on the left side in a `with` expression

Reply:
I see your point! Switched over to if/else to prevent obfuscating the logic.